modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/util/v1_0/AttachmentUtil.java

```diff
@@ -265,10 +269,20 @@ public static CPAttachmentFileEntry addOrUpdateCPAttachmentFileEntry(
 
 		DateConfig expirationDateConfig = new DateConfig(expirationCalendar);
 
+		ServiceContext cloneServiceContext =
+			(ServiceContext)serviceContext.clone();
+
+		ServiceContext cpAttachmentFileEntryServiceContext =
+			ServiceContextFactory.getInstance(
+				CPAttachmentFileEntry.class.getName(), httpServletRequest);
+
+		cloneServiceContext.setWorkflowAction(
+			cpAttachmentFileEntryServiceContext.getWorkflowAction());
+
```

Comment:
This makes no sense you can set the WF action directly in the cloned one, it will be always the default one

Reply:
Hey @andrea-ale-sbarra , I interpreted your statement in the following two ways; I will clarify both:
1.
```
		cloneServiceContext.setWorkflowAction(
			ServiceContextFactory.getInstance(
				CPAttachmentFileEntry.class.getName(), httpServletRequest
			).getWorkflowAction());

```
We cannot do this due to SourceFormat complaints.

2. The clone needs to happen because we need to use all the information from the actual serviceContext; only thing that needs to change for AttachmentUtil purposes is that we need to get the correct workflowAction from the servieContext of CPAttachmentFileEntry, the serviceContext that is passed in will have workflowACtion of CPDefinition.class.getName(). Also, we are cloning because we do not want to alter the workflowAction of the original serviceContext.

Reply:
```suggestion
		ServiceContext cloneServiceContext =
			(ServiceContext)serviceContext.clone();
		
		cloneServiceContext.setWorkflowAction(WorkflowConstants.ACTION_PUBLISH);

```

Reply:
Because if you see what's doing in the ServiceContextFactory that you used for workflowactionis as follow:

```
// Workflow

		serviceContext.setWorkflowAction(
			ParamUtil.getInteger(
				httpServletRequest, "workflowAction",
				WorkflowConstants.ACTION_PUBLISH));
```

And since in this headless request you for sure don't have the workflowAction attribute it will fallback to the default that is:

🎉 

WorkflowConstants.ACTION_PUBLISH

🎉

Reply:
@andrea-ale-sbarra , thank you for the clarification. I see exactly what you mean now. I should have been more observant on the default value set in the ServiceContextFactory.